src/test_deque.py

```diff
@@ -0,0 +1,333 @@
+    ('¥£', '¥')
+]
+
+PEEK_TABLE = [
+    ('1', '1'),
+    ('string', 's'),
+    ('$%^', '$'),
+    ([1, 2, 3], 1),
+]
+
+PEEKLEFT_TABLE = [
+    ('1', '1'),
+    ('string', 'g'),
+    ('$%^', '^'),
+    ([1, 2, 3], 3),
+]
+
+
+# Tests for __init__()
+
+
+def test_init_head_None():
+    """Test whether the head_node is None when initiated an empty Deque."""
+    deq = Deque()
+    assert not deq._doublyLL.head_node
+
+
+def test_init_tail_None():
+    """Test whether the tail_node is None when initiated an empty Deque."""
+    deq = Deque()
+    assert not deq._doublyLL.tail_node
+
+
+@pytest.mark.parametrize('iterable, length', INIT_TABLE1)
+def test_init_length(iterable, length):
+    """Test whether the length is right for an instance of Deque."""
+    deq = Deque(iterable)
+    assert len(deq) == length
+
+
+@pytest.mark.parametrize('iterable, head', INIT_TABLE2)
+def test_init_head(iterable, head):
+    """Test whether the head_node is rigth when initiate Deque(iterable)."""
+    deq = Deque(iterable)
+    assert deq._doublyLL.head_node.value == head
+
+
+@pytest.mark.parametrize('iterable, tail', INIT_TABLE3)
+def test_init_tail(iterable, tail):
+    """Test whether the tail_node is None when initiated an empty Deque."""
+    deq = Deque(iterable)
+    assert deq._doublyLL.tail_node.value == tail
+
+
+# Tests for .append()
+
+
+@pytest.mark.parametrize('val', APPEND_TABLE1)
+def test_append_tail_deq_1(val, deq_1):
+    """Test whether the tail_node.value is the appended value."""
+    deq_1.append(val)
+    assert deq_1._doublyLL.tail_node.value == val
+
+
+@pytest.mark.parametrize('iterable, val, result', APPEND_TABLE2)
+def test_append_size1_deq_2(iterable, val, result):
+    """Test whether len of the deque is right after a value was appended."""
+    deq = Deque(iterable)
+    deq.append(val)
+    assert len(deq) == result
+
+
+@pytest.mark.parametrize('iterable, val, result', APPEND_TABLE2)
+def test_append_size2_deq3(iterable, val, result):
+    """
+    Test whether len of the deque is right after calling
+    .append(val) - .pop() -.append(val) on a deque.
+    """
+    deq = Deque(iterable)
+    deq.append(val)
+    deq.pop()
+    deq.append(val)
+    assert len(deq) == result
+
+
+# Tests for .appendleft()
+
+
+@pytest.mark.parametrize('val', APPEND_TABLE1)
+def test_appendleft_tail_deq_1(val, deq_1):
+    """
+    Test whether the head_node.value is the value
+     that was passed in deq.appendleft(val).
+     """
+    deq_1.appendleft(val)
+    assert deq_1._doublyLL.head_node.value == val
+
+
+@pytest.mark.parametrize('iterable, val, result', APPEND_TABLE2)
+def test_appendleft_size_deq_2(iterable, val, result):
+    """
+    Test whether len of the deque is right after
+    calling deq.appendleft(val).
+    """
+    deq = Deque(iterable)
+    deq.appendleft(val)
+    assert len(deq) == result
+
+
+@pytest.mark.parametrize('iterable, val, result', APPEND_TABLE2)
+def test_appendleft_size_deq3(iterable, val, result):
```

Comment:
Good test to make sure your data structure still works after you mess around with a few different methods.